Oscar tests crash on Julia 1.4.2 on Ubuntu

I am getting type inference and other crashes in Julia-1.4.2 when running the Oscar tests.

```
julia> Pkg.test("Oscar")
    Testing Oscar
Status `/tmp/jl_DaghMs/Manifest.toml`
  [c3fe647b] AbstractAlgebra v0.9.2
  [9e28174c] BinDeps v1.0.1
  [b99e7846] BinaryProvider v0.5.10
  [631607c0] CMake v1.2.0
  [1f15a43c] CxxWrap v0.10.2
  [c863536a] GAP v0.4.1
  [3e1990a7] Hecke v0.8.3
  [682c06a0] JSON v0.21.0
  [472f376f] LoadFlint v0.1.3
  [1914dd2f] MacroTools v0.5.5
  [2edaba10] Nemo v0.17.5
  [f1435218] Oscar v0.3.0 [`~/.julia/dev/Oscar`]
  [69de0a69] Parsers v1.0.5
  [d720cf60] Polymake v0.4.2
  [ae029012] Requires v1.0.1
  [bcd08a7b] Singular v0.3.1
  [30578b45] URIParser v0.4.1
  [3eaa8342] libcxxwrap_julia_jll v0.7.1+1
  [2a0f44e3] Base64
  [ade2ca70] Dates
  [8ba89e20] Distributed
  [b77e0a4c] InteractiveUtils
  [76f85450] LibGit2
  [8f399da3] Libdl
  [37e2e46d] LinearAlgebra
  [56ddb016] Logging
  [d6f4376e] Markdown
  [a63ad114] Mmap
  [44cfe95a] Pkg
  [de0858da] Printf
  [9abbd945] Profile
  [3fa0cd96] REPL
  [9a3f8284] Random
  [ea8e919c] SHA
  [9e88b42a] Serialization
  [6462fe0b] Sockets
  [2f01184e] SparseArrays
  [10745b16] Statistics
  [8dfed614] Test
  [cf7118a7] UUIDs
  [4ec0a83e] Unicode
Test Summary:         | Pass  Total
Rings.ZZ.constructors |   14     14
Test Summary:       | Pass  Total
Rings.ZZ.properties |   41     41
Test Summary:       | Pass  Total
Rings.ZZ.arithmetic |   23     23
Test Summary:       | Pass  Total
Rings.ZZ.comparison |   50     50
Test Summary:     | Pass  Total
Rings.ZZ.divexact |   25     25
Test Summary:     | Pass  Total
Rings.ZZ.powering |   34     34
Test Summary:               | Pass  Total
Rings.ZZ.euclidean_division |   96     96
Test Summary:       | Pass  Total
Rings.ZZ.conversion |   17     17
Test Summary: | Pass  Total
Rings.ZZ.gcd  | 1205   1205
Test Summary:  | Pass  Total
Rings.ZZ.roots |   27     27
Test Summary:          | Pass  Total
Rings.ZZ.factorisation |   18     18
Test Summary:          | Pass  Total
Rings.ZZ.combinatorial |   91     91
Test Summary:               | Pass  Total
Rings.ZZ.number_theoretical |  103    103
Test Summary:         | Pass  Total
Rings.QQ.constructors |   68     68
Test Summary:       | Pass  Total
Rings.QQ.properties |   70     70
Test Summary:       | Pass  Total
Rings.QQ.arithmetic |   66     66
Test Summary:       | Pass  Total
Rings.QQ.comparison |   23     23
Test Summary:     | Pass  Total
Rings.QQ.divexact |   22     22
Test Summary:     | Pass  Total
Rings.QQ.powering |   23     23
Test Summary:        | Pass  Total
Polynomial Orderings |   18     18
Test Summary:           | Pass  Total
Polymake.factorisations |    2      2
Internal error: encountered unexpected error in runtime:
MethodError(f=typeof(Base.string)(), args=(Expr(:<:, :t, :r),), world=0x0000000000000f21)
jl_method_error_bare at /buildworker/worker/package_linux64/build/src/gf.c:1719
jl_method_error at /buildworker/worker/package_linux64/build/src/gf.c:1737
jl_lookup_generic_ at /buildworker/worker/package_linux64/build/src/gf.c:2298 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2319
macro expansion at ./error.jl:223 [inlined]
limit_type_size at ./compiler/typelimits.jl:23
jfptr_limit_type_size_4439.clone_1 at /home/wbhart/julia-1.4.2/lib/julia/sys.so (unknown line)
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2145 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2323
abstract_call_method at ./compiler/abstractinterpretation.jl:359
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1227
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1213
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1295
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1227
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1227
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1227
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1213
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1227
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1227
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1295
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1227
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1227
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1227
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1227
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1213
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1213
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1213
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1213
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1227
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1227
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1227
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1213
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:488
abstract_call_method at ./compiler/abstractinterpretation.jl:404
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:101
abstract_call_known at ./compiler/abstractinterpretation.jl:873
abstract_call at ./compiler/abstractinterpretation.jl:895
abstract_call at ./compiler/abstractinterpretation.jl:880
abstract_eval at ./compiler/abstractinterpretation.jl:974
typeinf_local at ./compiler/abstractinterpretation.jl:1227
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1283
typeinf at ./compiler/typeinfer.jl:12
typeinf_ext at ./compiler/typeinfer.jl:574
typeinf_ext at ./compiler/typeinfer.jl:605
jfptr_typeinf_ext_1.clone_1 at /home/wbhart/julia-1.4.2/lib/julia/sys.so (unknown line)
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2145 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2323
jl_apply at /buildworker/worker/package_linux64/build/src/julia.h:1700 [inlined]
jl_type_infer at /buildworker/worker/package_linux64/build/src/gf.c:213
jl_compile_method_internal at /buildworker/worker/package_linux64/build/src/gf.c:1888
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2154 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2323
jl_apply at /buildworker/worker/package_linux64/build/src/julia.h:1700 [inlined]
do_call at /buildworker/worker/package_linux64/build/src/interpreter.c:369
eval_value at /buildworker/worker/package_linux64/build/src/interpreter.c:458
eval_body at /buildworker/worker/package_linux64/build/src/interpreter.c:680
eval_body at /buildworker/worker/package_linux64/build/src/interpreter.c:744
eval_body at /buildworker/worker/package_linux64/build/src/interpreter.c:744
jl_interpret_toplevel_thunk at /buildworker/worker/package_linux64/build/src/interpreter.c:911
macro expansion at /home/wbhart/.julia/dev/Oscar/test/Polymake/nmbthy-test.jl:16 [inlined]
macro expansion at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Test/src/Test.jl:1113 [inlined]
top-level scope at /home/wbhart/.julia/dev/Oscar/test/Polymake/nmbthy-test.jl:13
jl_toplevel_eval_flex at /buildworker/worker/package_linux64/build/src/toplevel.c:819
jl_parse_eval_all at /buildworker/worker/package_linux64/build/src/ast.c:872
jl_load at /buildworker/worker/package_linux64/build/src/toplevel.c:877
include at ./client.jl:439
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2145 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2323
jl_apply at /buildworker/worker/package_linux64/build/src/julia.h:1700 [inlined]
do_call at /buildworker/worker/package_linux64/build/src/interpreter.c:369
eval_value at /buildworker/worker/package_linux64/build/src/interpreter.c:458
eval_stmt_value at /buildworker/worker/package_linux64/build/src/interpreter.c:409 [inlined]
eval_body at /buildworker/worker/package_linux64/build/src/interpreter.c:817
jl_interpret_toplevel_thunk at /buildworker/worker/package_linux64/build/src/interpreter.c:911
top-level scope at none:7
jl_toplevel_eval_flex at /buildworker/worker/package_linux64/build/src/toplevel.c:819
jl_parse_eval_all at /buildworker/worker/package_linux64/build/src/ast.c:872
jl_load at /buildworker/worker/package_linux64/build/src/toplevel.c:877
include at ./client.jl:439
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2145 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2323
jl_apply at /buildworker/worker/package_linux64/build/src/julia.h:1700 [inlined]
do_call at /buildworker/worker/package_linux64/build/src/interpreter.c:369
eval_value at /buildworker/worker/package_linux64/build/src/interpreter.c:458
eval_stmt_value at /buildworker/worker/package_linux64/build/src/interpreter.c:409 [inlined]
eval_body at /buildworker/worker/package_linux64/build/src/interpreter.c:817
jl_interpret_toplevel_thunk at /buildworker/worker/package_linux64/build/src/interpreter.c:911
top-level scope at none:6
jl_toplevel_eval_flex at /buildworker/worker/package_linux64/build/src/toplevel.c:819
jl_toplevel_eval_flex at /buildworker/worker/package_linux64/build/src/toplevel.c:769
jl_toplevel_eval_in at /buildworker/worker/package_linux64/build/src/toplevel.c:848
eval at ./boot.jl:331
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2145 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2323
exec_options at ./client.jl:264
_start at ./client.jl:484
jfptr__start_2075.clone_1 at /home/wbhart/julia-1.4.2/lib/julia/sys.so (unknown line)
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2145 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2323
jl_apply at /buildworker/worker/package_linux64/build/ui/../src/julia.h:1700 [inlined]
true_main at /buildworker/worker/package_linux64/build/ui/repl.c:96
main at /buildworker/worker/package_linux64/build/ui/repl.c:217
__libc_start_main at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
_start at /home/wbhart/julia-1.4.2/bin/julia (unknown line)
Test Summary:          | Pass  Total
Polymake.norm_equation |    1      1
Test Summary:                   | Pass  Total
GAPGroups_interface_conformance |  183    183
Iteration: Error During Test at /home/wbhart/.julia/dev/Oscar/test/Groups/testing.jl:178
  Got exception outside of a @test
  failed to convert GapObj to any known type:
   GAP: ()
  Stacktrace:
   [1] gap_to_julia(::Main.ForeignGAP.MPtr; recursive::Bool) at /home/wbhart/.julia/packages/GAP/7sFvk/src/gap_to_julia.jl:311
   [2] #gap_to_julia#9 at /home/wbhart/.julia/packages/GAP/7sFvk/src/gap_to_julia.jl:57 [inlined]
   [3] #gap_to_julia#7 at /home/wbhart/.julia/packages/GAP/7sFvk/src/gap_to_julia.jl:53 [inlined]
   [4] #13 at /home/wbhart/.julia/packages/GAP/7sFvk/src/gap_to_julia.jl:163 [inlined]
   [5] get!(::GAP.var"#13#14"{Any,IdDict{Any,Any},Main.ForeignGAP.MPtr}, ::IdDict{Any,Any}, ::Any) at ./abstractdict.jl:663
   [6] gap_to_julia(::Type{Array{Any,1}}, ::Main.ForeignGAP.MPtr, ::IdDict{Any,Any}; recursive::Bool) at /home/wbhart/.julia/packages/GAP/7sFvk/src/gap_to_julia.jl:162
   [7] gap_to_julia(::Main.ForeignGAP.MPtr; recursive::Bool) at /home/wbhart/.julia/packages/GAP/7sFvk/src/gap_to_julia.jl:309
   [8] gap_to_julia(::Main.ForeignGAP.MPtr) at /home/wbhart/.julia/packages/GAP/7sFvk/src/gap_to_julia.jl:299
   [9] elements(::PermGroup) at /home/wbhart/.julia/dev/Oscar/src/Groups/GAPGroups.jl:67
   [10] macro expansion at /home/wbhart/.julia/dev/Oscar/test/Groups/testing.jl:181 [inlined]
   [11] macro expansion at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Test/src/Test.jl:1113 [inlined]
   [12] top-level scope at /home/wbhart/.julia/dev/Oscar/test/Groups/testing.jl:179
   [13] include(::String) at ./client.jl:439
   [14] top-level scope at /home/wbhart/.julia/dev/Oscar/test/Groups/runtests.jl:4
   [15] include(::String) at ./client.jl:439
   [16] top-level scope at none:8
   [17] include(::String) at ./client.jl:439
   [18] top-level scope at none:6
   [19] eval(::Module, ::Any) at ./boot.jl:331
   [20] exec_options(::Base.JLOptions) at ./client.jl:264
   [21] _start() at ./client.jl:484

Test Summary: | Error  Total
Iteration     |     1      1
ERROR: LoadError: LoadError: LoadError: Some tests did not pass: 0 passed, 0 failed, 1 errored, 0 broken.
in expression starting at /home/wbhart/.julia/dev/Oscar/test/Groups/testing.jl:178
in expression starting at /home/wbhart/.julia/dev/Oscar/test/Groups/runtests.jl:4
in expression starting at /home/wbhart/.julia/dev/Oscar/test/runtests.jl:8
ERROR: Package Oscar errored during testing
Stacktrace:
 [1] pkgerror(::String, ::Vararg{String,N} where N) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Pkg/src/Types.jl:53
 [2] test(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}; coverage::Bool, julia_args::Cmd, test_args::Cmd, test_fn::Nothing) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Pkg/src/Operations.jl:1510
 [3] test(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}; coverage::Bool, test_fn::Nothing, julia_args::Cmd, test_args::Cmd, kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Pkg/src/API.jl:316
 [4] test(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Pkg/src/API.jl:303
 [5] #test#68 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Pkg/src/API.jl:297 [inlined]
 [6] test at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Pkg/src/API.jl:297 [inlined]
 [7] #test#67 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Pkg/src/API.jl:296 [inlined]
 [8] test at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Pkg/src/API.jl:296 [inlined]
 [9] test(::String; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Pkg/src/API.jl:295
 [10] test(::String) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Pkg/src/API.jl:295
 [11] top-level scope at REPL[8]:1
```

I don't think this is related to the Polymake Integer issue, but more some type inference bug in julia, maybe similar to https://github.com/JuliaLang/julia/issues/23786. Maybe stemming from Hecke or Nemo types?

It can be triggered in the REPL with the following. And it triggers before any actual code is run, a `@show` in the first line of `norm_equation_fac_elem` will print after the error.
```
julia> using Oscar
 -----    -----    -----      -      -----   
|     |  |     |  |     |    | |    |     |  
|     |  |        |         |   |   |     |  
|     |   -----   |        |     |  |-----   
|     |        |  |        |-----|  |   |    
|     |  |     |  |     |  |     |  |    |   
 -----    -----    -----   -     -  -     -  

...combining (and extending) ANTIC, GAP, Polymake and Singular
Version 0.3.0-dev ... 
 ... which comes with absolutely no warranty whatsoever
Type: '?Oscar' for more information
(c) 2019-2020 by The Oscar Development Team

julia> k, a = wildanger_field(3, 13)
(Number field over Rational Field with defining polynomial x^3-13*x^2+13*x-13, _$)

julia> zk = maximal_order(k)
Maximal order of Number field over Rational Field with defining polynomial x^3-13*x^2+13*x-13 
with basis nf_elem[1, _$, 1//2*_$^2 + 1//2]

julia> na = norm(rand(zk, 1:10))
17991

julia> s = norm_equation(zk, na)
Internal error: encountered unexpected error in runtime:
MethodError(f=typeof(Base.string)(), args=(Expr(:<:, :t, :r),), world=0x0000000000000f21)
jl_method_error_bare at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/src/gf.c:1719
jl_method_error at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/src/gf.c:1737
...
jl_toplevel_eval_in at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/src/toplevel.c:848
eval at ./boot.jl:331
eval_user_input at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:86
macro expansion at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:118 [inlined]
#26 at ./task.jl:358
jl_apply at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/src/julia.h:1700 [inlined]
start_task at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/src/task.c:687
unknown function (ip: (nil))

1-element Array{NfAbsOrdElem{AnticNumberField,nf_elem},1}:
 _$^2 + 6*_$ + 4
```

Same error happens for this:
```
julia> norm_equation_fac_elem(zk,fmpz(na))
Internal error: encountered unexpected error in runtime:
MethodError(f=typeof(Base.string)(), args=(Expr(:<:, :t, :r),), world=0x0000000000000f21)
jl_method_error_bare at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/src/gf.c:1719
...
```

Interestingly it does not appear if the optional Bool is specified explicitly:
```
julia> norm_equation_fac_elem(zk,fmpz(na),abs=false)
1-element Array{FacElem{nf_elem,AnticNumberField},1}:
 Factored element with data
Dict{nf_elem,fmpz}(-41//5997*_$^2 + 554//5997*_$ - 256//1999 => -1,-1 => 1)

```

Also julia `@code_warntype` fails with an AssertionError. julia seems to be very deep in the type inference and then fail because some type is too complex? (the assertion is here: https://github.com/JuliaLang/julia/blob/master/base/compiler/typelimits.jl#L23)
```
julia> @code_warntype norm_equation_fac_elem(zk,fmpz(na))
ERROR: AssertionError: t <: r
Stacktrace:
 [1] limit_type_size(::Any, ::Any, ::Any, ::Int64, ::Int64) at ./compiler/typelimits.jl:23
 [2] abstract_call_method(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:359
 [3] abstract_call_gf_by_type(::Any, ::Array{Any,1}, ::Any, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:101
 [4] abstract_call_known(::Any, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:873
 [5] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:895
 [6] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:880
 [7] abstract_eval(::Any, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:974
 [8] typeinf_local(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1227
 [9] typeinf_nocycle(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1283
 [10] typeinf(::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:12
 [11] typeinf_edge(::Method, ::Any, ::Core.SimpleVector, ::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:488
 [12] abstract_call_method(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:404
 [13] abstract_call_gf_by_type(::Any, ::Array{Any,1}, ::Any, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:101
 [14] abstract_call_known(::Any, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:873
 [15] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:895
 [16] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:880
 [17] abstract_eval(::Any, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:974
 [18] typeinf_local(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1213
 [19] typeinf_nocycle(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1295
 [20] typeinf(::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:12
 [21] typeinf_edge(::Method, ::Any, ::Core.SimpleVector, ::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:488
 [22] abstract_call_method(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:404
 [23] abstract_call_gf_by_type(::Any, ::Array{Any,1}, ::Any, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:101
 [24] abstract_call_known(::Any, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:873
 [25] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:895
 [26] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:880
 [27] abstract_eval(::Any, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:974
 [28] typeinf_local(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1227
 [29] typeinf_nocycle(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1283
 ... (the last 10 lines are repeated 2 more times)
 [50] typeinf(::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:12
 [51] typeinf_edge(::Method, ::Any, ::Core.SimpleVector, ::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:488
 [52] abstract_call_method(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:404
 [53] abstract_call_gf_by_type(::Any, ::Array{Any,1}, ::Any, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:101
 [54] abstract_call_known(::Any, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:873
 [55] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:895
 [56] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:880
 [57] abstract_eval(::Any, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:974
 [58] typeinf_local(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1213
 [59] typeinf_nocycle(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1283
 [60] typeinf(::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:12
 [61] typeinf_edge(::Method, ::Any, ::Core.SimpleVector, ::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:488
 [62] abstract_call_method(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:404
 [63] abstract_call_gf_by_type(::Any, ::Array{Any,1}, ::Any, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:101
 [64] abstract_call_known(::Any, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:873
 [65] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:895
 [66] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:880
 [67] abstract_eval(::Any, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:974
 [68] typeinf_local(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1227
 ... (the last 10 lines are repeated 1 more time)
 [79] typeinf_nocycle(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1295
 [80] typeinf(::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:12
 [81] typeinf_edge(::Method, ::Any, ::Core.SimpleVector, ::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:488
 [82] abstract_call_method(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:404
 [83] abstract_call_gf_by_type(::Any, ::Array{Any,1}, ::Any, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:101
 [84] abstract_call_known(::Any, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:873
 [85] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:895
 [86] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:880
 [87] abstract_eval(::Any, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:974
 [88] typeinf_local(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1227
 [89] typeinf_nocycle(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1283
 ... (the last 10 lines are repeated 3 more times)
 [120] typeinf(::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:12
 [121] typeinf_edge(::Method, ::Any, ::Core.SimpleVector, ::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:488
 [122] abstract_call_method(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:404
 [123] abstract_call_gf_by_type(::Any, ::Array{Any,1}, ::Any, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:101
 [124] abstract_call_known(::Any, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:873
 [125] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:895
 [126] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:880
 [127] abstract_eval(::Any, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:974
 [128] typeinf_local(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1213
 [129] typeinf_nocycle(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1283
 ... (the last 10 lines are repeated 3 more times)
 [160] typeinf(::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:12
 [161] typeinf_edge(::Method, ::Any, ::Core.SimpleVector, ::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:488
 [162] abstract_call_method(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:404
 [163] abstract_call_gf_by_type(::Any, ::Array{Any,1}, ::Any, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:101
 [164] abstract_call_known(::Any, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:873
 [165] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:895
 [166] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:880
 [167] abstract_eval(::Any, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:974
 [168] typeinf_local(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1227
 [169] typeinf_nocycle(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1283
 ... (the last 10 lines are repeated 2 more times)
 [190] typeinf(::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:12
 [191] typeinf_code(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.Params) at ./compiler/typeinfer.jl:515
 [192] code_typed(::Any, ::Any; optimize::Bool, debuginfo::Symbol, world::UInt64, params::Core.Compiler.Params) at ./reflection.jl:1107
 [193] code_warntype(::Base.TTY, ::Any, ::Any; debuginfo::Symbol, optimize::Bool) at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/usr/share/julia/stdlib/v1.4/InteractiveUtils/src/codeview.jl:37
 [194] code_warntype(::Base.TTY, ::Any, ::Any) at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/usr/share/julia/stdlib/v1.4/InteractiveUtils/src/codeview.jl:35
 [195] code_warntype(::Any, ::Any; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/usr/share/julia/stdlib/v1.4/InteractiveUtils/src/codeview.jl:61
 [196] code_warntype(::Any, ::Any) at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/usr/share/julia/stdlib/v1.4/InteractiveUtils/src/codeview.jl:61
```

On Fri, Jun 19, 2020 at 04:09:00AM -0700, Benjamin Lorenz wrote:
> I don't think this is related to the Polymake Integer issue, but more some type inference bug in julia, maybe similar to https://github.com/JuliaLang/julia/issues/23786. Maybe stemming from Hecke or Nemo types?
> 
> It can be triggered in the REPL with the following. And it triggers before any actual code is run, a `@show` in the first line of `norm_equation_fac_elem` will print after the error.
> ```
it's the mixture.

Hecke "alone" is fine:
In Oscar
Hecke.build()
is fine
Oscar.build()
the errors.

Indeed, some type seems to be too complex.
> julia> using Oscar
>  -----    -----    -----      -      -----   
> |     |  |     |  |     |    | |    |     |  
> |     |  |        |         |   |   |     |  
> |     |   -----   |        |     |  |-----   
> |     |        |  |        |-----|  |   |    
> |     |  |     |  |     |  |     |  |    |   
>  -----    -----    -----   -     -  -     -  
> 
> ...combining (and extending) ANTIC, GAP, Polymake and Singular
> Version 0.3.0-dev ... 
>  ... which comes with absolutely no warranty whatsoever
> Type: '?Oscar' for more information
> (c) 2019-2020 by The Oscar Development Team
> 
> julia> k, a = wildanger_field(3, 13)
> (Number field over Rational Field with defining polynomial x^3-13*x^2+13*x-13, _$)
> 
> julia> zk = maximal_order(k)
> Maximal order of Number field over Rational Field with defining polynomial x^3-13*x^2+13*x-13 
> with basis nf_elem[1, _$, 1//2*_$^2 + 1//2]
> 
> julia> na = norm(rand(zk, 1:10))
> 17991
> 
> julia> s = norm_equation(zk, na)
> Internal error: encountered unexpected error in runtime:
> MethodError(f=typeof(Base.string)(), args=(Expr(:<:, :t, :r),), world=0x0000000000000f21)
> jl_method_error_bare at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/src/gf.c:1719
> jl_method_error at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/src/gf.c:1737
> ...
> jl_toplevel_eval_in at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/src/toplevel.c:848
> eval at ./boot.jl:331
> eval_user_input at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:86
> macro expansion at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:118 [inlined]
> #26 at ./task.jl:358
> jl_apply at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/src/julia.h:1700 [inlined]
> start_task at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/src/task.c:687
> unknown function (ip: (nil))
> 
> 1-element Array{NfAbsOrdElem{AnticNumberField,nf_elem},1}:
>  _$^2 + 6*_$ + 4
> ```
> 
> Same error happens for this:
> ```
> julia> norm_equation_fac_elem(zk,fmpz(na))
> Internal error: encountered unexpected error in runtime:
> MethodError(f=typeof(Base.string)(), args=(Expr(:<:, :t, :r),), world=0x0000000000000f21)
> jl_method_error_bare at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/src/gf.c:1719
> ...
> ```
> 
> Interestingly it does not appear if the optional Bool is specified explicitly:
> ```
> julia> norm_equation_fac_elem(zk,fmpz(na),abs=false)
> 1-element Array{FacElem{nf_elem,AnticNumberField},1}:
>  Factored element with data
> Dict{nf_elem,fmpz}(-41//5997*_$^2 + 554//5997*_$ - 256//1999 => -1,-1 => 1)
> 
> ```
> 
> Also julia `@code_warntype` fails with an AssertionError. julia seems to be very deep in the type inference and then fail because some type is too complex? (the assertion is here: https://github.com/JuliaLang/julia/blob/master/base/compiler/typelimits.jl#L23)
> ```
> julia> @code_warntype norm_equation_fac_elem(zk,fmpz(na))
> ERROR: AssertionError: t <: r
> Stacktrace:
>  [1] limit_type_size(::Any, ::Any, ::Any, ::Int64, ::Int64) at ./compiler/typelimits.jl:23
>  [2] abstract_call_method(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:359
>  [3] abstract_call_gf_by_type(::Any, ::Array{Any,1}, ::Any, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:101
>  [4] abstract_call_known(::Any, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:873
>  [5] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:895
>  [6] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:880
>  [7] abstract_eval(::Any, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:974
>  [8] typeinf_local(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1227
>  [9] typeinf_nocycle(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1283
>  [10] typeinf(::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:12
>  [11] typeinf_edge(::Method, ::Any, ::Core.SimpleVector, ::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:488
>  [12] abstract_call_method(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:404
>  [13] abstract_call_gf_by_type(::Any, ::Array{Any,1}, ::Any, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:101
>  [14] abstract_call_known(::Any, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:873
>  [15] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:895
>  [16] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:880
>  [17] abstract_eval(::Any, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:974
>  [18] typeinf_local(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1213
>  [19] typeinf_nocycle(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1295
>  [20] typeinf(::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:12
>  [21] typeinf_edge(::Method, ::Any, ::Core.SimpleVector, ::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:488
>  [22] abstract_call_method(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:404
>  [23] abstract_call_gf_by_type(::Any, ::Array{Any,1}, ::Any, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:101
>  [24] abstract_call_known(::Any, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:873
>  [25] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:895
>  [26] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:880
>  [27] abstract_eval(::Any, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:974
>  [28] typeinf_local(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1227
>  [29] typeinf_nocycle(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1283
>  ... (the last 10 lines are repeated 2 more times)
>  [50] typeinf(::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:12
>  [51] typeinf_edge(::Method, ::Any, ::Core.SimpleVector, ::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:488
>  [52] abstract_call_method(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:404
>  [53] abstract_call_gf_by_type(::Any, ::Array{Any,1}, ::Any, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:101
>  [54] abstract_call_known(::Any, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:873
>  [55] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:895
>  [56] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:880
>  [57] abstract_eval(::Any, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:974
>  [58] typeinf_local(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1213
>  [59] typeinf_nocycle(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1283
>  [60] typeinf(::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:12
>  [61] typeinf_edge(::Method, ::Any, ::Core.SimpleVector, ::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:488
>  [62] abstract_call_method(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:404
>  [63] abstract_call_gf_by_type(::Any, ::Array{Any,1}, ::Any, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:101
>  [64] abstract_call_known(::Any, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:873
>  [65] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:895
>  [66] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:880
>  [67] abstract_eval(::Any, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:974
>  [68] typeinf_local(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1227
>  ... (the last 10 lines are repeated 1 more time)
>  [79] typeinf_nocycle(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1295
>  [80] typeinf(::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:12
>  [81] typeinf_edge(::Method, ::Any, ::Core.SimpleVector, ::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:488
>  [82] abstract_call_method(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:404
>  [83] abstract_call_gf_by_type(::Any, ::Array{Any,1}, ::Any, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:101
>  [84] abstract_call_known(::Any, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:873
>  [85] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:895
>  [86] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:880
>  [87] abstract_eval(::Any, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:974
>  [88] typeinf_local(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1227
>  [89] typeinf_nocycle(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1283
>  ... (the last 10 lines are repeated 3 more times)
>  [120] typeinf(::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:12
>  [121] typeinf_edge(::Method, ::Any, ::Core.SimpleVector, ::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:488
>  [122] abstract_call_method(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:404
>  [123] abstract_call_gf_by_type(::Any, ::Array{Any,1}, ::Any, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:101
>  [124] abstract_call_known(::Any, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:873
>  [125] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:895
>  [126] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:880
>  [127] abstract_eval(::Any, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:974
>  [128] typeinf_local(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1213
>  [129] typeinf_nocycle(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1283
>  ... (the last 10 lines are repeated 3 more times)
>  [160] typeinf(::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:12
>  [161] typeinf_edge(::Method, ::Any, ::Core.SimpleVector, ::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:488
>  [162] abstract_call_method(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:404
>  [163] abstract_call_gf_by_type(::Any, ::Array{Any,1}, ::Any, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:101
>  [164] abstract_call_known(::Any, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:873
>  [165] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState, ::Int64) at ./compiler/abstractinterpretation.jl:895
>  [166] abstract_call(::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:880
>  [167] abstract_eval(::Any, ::Array{Any,1}, ::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:974
>  [168] typeinf_local(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1227
>  [169] typeinf_nocycle(::Core.Compiler.InferenceState) at ./compiler/abstractinterpretation.jl:1283
>  ... (the last 10 lines are repeated 2 more times)
>  [190] typeinf(::Core.Compiler.InferenceState) at ./compiler/typeinfer.jl:12
>  [191] typeinf_code(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.Params) at ./compiler/typeinfer.jl:515
>  [192] code_typed(::Any, ::Any; optimize::Bool, debuginfo::Symbol, world::UInt64, params::Core.Compiler.Params) at ./reflection.jl:1107
>  [193] code_warntype(::Base.TTY, ::Any, ::Any; debuginfo::Symbol, optimize::Bool) at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/usr/share/julia/stdlib/v1.4/InteractiveUtils/src/codeview.jl:37
>  [194] code_warntype(::Base.TTY, ::Any, ::Any) at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/usr/share/julia/stdlib/v1.4/InteractiveUtils/src/codeview.jl:35
>  [195] code_warntype(::Any, ::Any; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/usr/share/julia/stdlib/v1.4/InteractiveUtils/src/codeview.jl:61
>  [196] code_warntype(::Any, ::Any) at /var/tmp/portage/dev-lang/julia-1.4.2/work/julia-1.4.2/usr/share/julia/stdlib/v1.4/InteractiveUtils/src/codeview.jl:61
> ```
> 
> -- 
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/oscar-system/Oscar.jl/issues/104#issuecomment-646577624

--- a/src/Polymake/NmbThy.jl
+++ b/src/Polymake/NmbThy.jl
@@ -3,19 +3,19 @@ export norm_equation_fac_elem, norm_equation, irreducibles, factorisations
 function norm_equation_fac_elem(R::NfAbsOrd, k::fmpz; abs::Bool = false)
   @assert Hecke.ismaximal(R)
   lp = factor(k)
-  S = []
+  S = Tuple{Vector{Tuple{Hecke.ideal_type(R), Int}}, Vector{fmpz_mat}}[]
   for (p, k) = lp.fac
     P = prime_decomposition(R, p)
     s = solve_non_negative(matrix(FlintZZ, 1, length(P), [degree(x[1]) for x = P]), matrix(FlintZZ, 1, 1, [k]))
-    push!(S, (P, [view(s, i:i, 1:ncols(s)) for i=1:nrows(s)]))
+    push!(S, (P, fmpz_mat[view(s, i:i, 1:ncols(s)) for i=1:nrows(s)]))
   end
-  sol = []
+  sol = FacElem{nf_elem, AnticNumberField}[]
   for x in Base.Iterators.ProductIterator(Tuple(t[2] for t = S))
     I = ideal(R, 1)
     for i = 1:length(S)
-      I *= prod(S[i][1][j][1]^Int(x[i][j]) for j=1:length(S[i][1]))
+      I *= prod(Hecke.ideal_type(R)[S[i][1][j][1]^Int(x[i][j]) for j=1:length(S[i][1])])
     end
-    fl, g = Hecke.isprincipal_fac_elem(I)
+    fl, g = Hecke.isprincipal_fac_elem(I::Hecke.ideal_type(R))
     if fl
       push!(sol, g)
     end
@@ -23,12 +23,12 @@ function norm_equation_fac_elem(R::NfAbsOrd, k::fmpz; abs::Bool = false)
   if !abs
     u, mu = unit_group_fac_elem(R)
     i = findfirst(x -> norm(mu(x)) == -1, gens(u))
-    ns = [norm(x) for x = sol]
+    ns = fmpq[norm(x) for x = sol]
     if i === nothing
       return [sol[i] for i in 1:length(sol) if ns[i] == k]
     end
     U = mu(u[i])
-    return [ ns[i] == k ? sol[i] : sol[i] * U for i = 1:length(sol)]
+    return FacElem{nf_elem, AnticNumberField}[ ns[i] == k ? sol[i] : sol[i] * U for i = 1:length(sol)]
   end
   return sol
 end
@@ -38,7 +38,7 @@ norm_equation_fac_elem(R::NfAbsOrd, k::Base.Integer; abs::Bool = false) =
 
 function norm_equation(R::NfAbsOrd, k::fmpz; abs::Bool = false)
   s = norm_equation_fac_elem(R, k, abs = abs)
-  return [R(evaluate(x)) for x = s]
+  return elem_type(R)[R(evaluate(x)) for x = s]
 end
 
 norm_equation(R::NfAbsOrd, k::Base.Integer; abs::Bool = false) = norm_equation(R, fmpz(k), abs = abs)

